### PR TITLE
Extension snapshot

### LINKED ISSUE #3503: Test if publishing works as expected
Recently we had several problems when modules publish unexpected artifacts. That lead to failures in downstream projects which depend on them, such as custom distributions.

We need to have automated tests for at least two separate scenarios:

- [ ] External extension can consume required artifacts to successfully build and test itself
- [ ] External library dependency can consume required artifacts to successfully build and test itself

Both external modules can live in our `examples` folder. The test itself should build the main project, publish it to local maven repository and then try to build-test example module, consuming dependencies from local maven repo.

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #3503

After this PR we always build examples against locally built and published snapshot.

In this PR examples does not use Sonatype snapshot repo, only `mavenCentral` and `mavenLocal`. This makes sure that when we want to test that the _current_ version of the our main repo publishes artifacts correctly, then we actually will test locally published versions and not something from Sonatype. The downside of that is that one cannot build examples until there is locally published version of the main project. This may be pretty big downside.

The better approach would be to use Sonatype snapshot repo by default but during publishing test somehow remove it and use only local repo. I see only one way to achieve that: via some shell script manipulating `build.gradle`, which is quite fragile IMO. All ideas are welcomed.

### CODE REVIEW

**Comment by iNikem** [comment]
I solved the previous problem. Now all builds except nightly will consume snapshots from Sonatype repo. Nightly will be forced to use locally published one.

File: examples/distro/build.gradle [hunk 1]
```diff
@@ -29,10 +29,8 @@ subprojects {
   }
 
   repositories {
-    maven {
-      url = uri("https://oss.sonatype.org/content/repositories/snapshots")
```
Comment (mateuszrzeszutek) [inline examples/distro/build.gradle:33]: Can we start using SNAPSHOTs in distro example too? (Separate PR?)

**Review by mateuszrzeszutek — APPROVED** [review]

File: examples/distro/build.gradle (hunk 1 above)
Comment (iNikem) [reply examples/distro/build.gradle:33]: Absolutely, will fix right away

Comment (iNikem) [reply examples/distro/build.gradle:33]: Turns out, this cannot be done until we fix publishing of `agent-for-testing`.

**Review by anuraaga — APPROVED** [review]

File: examples/distro/build.gradle (hunk 1 above)
Comment (iNikem) [reply examples/distro/build.gradle:33]: Done

**Review by mateuszrzeszutek — APPROVED** [review]

File: .github/scripts/local.init.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+addListener(object : BuildAdapter() {
+  override fun projectsEvaluated(gradle: Gradle) {
+    gradle.rootProject {
+      repositories {
+        mavenLocal()
+        removeIf { it.name == "sonatype" }
+      }
+    }
+  }
+})
```
Comment (trask) [inline .github/scripts/local.init.gradle.kts:10]: wow

**Review by trask — COMMENTED** [review]
nice, this is awesome

**Pull request merged**